server/resources/resourcesController.js

```diff
@@ -48,6 +49,8 @@ router
 			const { id: source } = req.user;
 			try {
 				const resource = await service.create({ ...req.body, source });
+				const validated_admin = await usersService.validateAdmin(source);
```

Comment:
❌ You don't actually need to validate the `source` user's status here, but it might not be obvious why:

- The `authOnly` middleware ensures that the `req.user` is definitely set, so it's guaranteed that the user object will be available to us.
- We don't set `req.user` explicitly anywhere, because that's done by 
the [Passport session](https://www.passportjs.org/concepts/authentication/sessions/) system.
- The `connect.sid` cookie which is used for authentication includes the user's ID:

    https://github.com/fikretellek/tech-products-demo/blob/d97ed9189c5d8a50670320c4e5f1a0c83a1113ea/server/auth/authController.js#L39

- When a request is received with a valid cookie, we get the appropriate vlaue for `req.user` from the service (`user` here is the ID, because that's what we serialized above):

    https://github.com/fikretellek/tech-products-demo/blob/d97ed9189c5d8a50670320c4e5f1a0c83a1113ea/server/auth/authController.js#L31-L37

- Which gets it directly from the database:

    https://github.com/fikretellek/tech-products-demo/blob/d97ed9189c5d8a50670320c4e5f1a0c83a1113ea/server/auth/authService.js#L3-L5

The cookie is validated using the session secret, so can't be tampered with. If we can trust the `id` in the `req.user`, we can trust the `is_admin` property too - the user cannot fake these (unlike the request body, query parameters or other headers).

Reply:
Authentication is always so complicated to me but It makes more sense now. I understand better now,  thank you very much @textbook 🙏